### ISSUE TITLE
false positive with method does not exist inside is_a($className, $interface, true);

### ISSUE THREAD

**Issue by WDD-Marv** [vimeo/psalm#897]
with late-static-binding, this should not be an issue:
1) use LSB to check if abstract class implements interface
2) use LSB to call method.

https://getpsalm.org/r/f7b6109ad0
```php
<?php

interface Foo {
  public static function DoFooThing();
}

abstract class Bar {
  public static function DooBarThing() : void {
    if (is_a(static::class, Foo::class, true)) {
      static::DoFooThing();
    }
  }
}

class Baz extends Bar implements Foo {
  public static function DoFooThing() : string {
    return '';
  }
}
```

versions:
* 6f96c66
* 0beeb87 aka 1.1.7

**Comment by muglug** [comment]
Thanks!

**Comment by WDD-Marv** [comment]
would be nice if this was also back-ported to the 1.x branch :D
